scipy/sparse/_index.py

```diff
@@ -276,9 +276,9 @@ def _unpack_index(index):
             'Indexing with sparse matrices is not supported '
             'except boolean indexing where matrix and index '
             'are equal shapes.')
-    if isinstance(row, np.ndarray) and row.dtype.kind == 'b':
+    if hasattr(row, 'ndim') and row.dtype.kind == 'b':
```

Comment:
Leave a comment here to explain why we're checking for `ndim` despite not accessing it. Or better, define a small helper function `_is_boolean_index` that performs these checks.

Reply:
Making a separate helper function will be a good idea.